localstack/services/awslambda/lambda_utils.py

```diff
@@ -288,3 +289,110 @@ def generate_lambda_arn(
         return f"arn:aws:lambda:{region}:{account_id}:function:{fn_name}:{qualifier}"
     else:
         return f"arn:aws:lambda:{region}:{account_id}:function:{fn_name}"
+
+
+class FilterCriteria(TypedDict):
+    Filters: List[Dict[str, any]]
+
+
+def parse_and_apply_numeric_filter(record_value: Dict, numeric_filter: List[Union[str, int]]):
+    if len(numeric_filter) % 2 > 0:
+        LOG.warn("Invalid numeric lambda filter given")
+        return True
+    all_comparisons = []
+    try:
+        # record_value is a dynamo DB entry in following form { datatype : value}
+        record_value = float(record_value.get("N"))
+    except (ValueError, TypeError):
+        LOG.warn(f"Could not convert Record {record_value} to a valid number value for filtering")
+        return True
+    for idx in range(0, len(numeric_filter), 2):
+
+        try:
+            if numeric_filter[idx] == ">" and record_value > float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            if numeric_filter[idx] == ">=" and record_value >= float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            if numeric_filter[idx] == "=" and record_value == float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            if numeric_filter[idx] == "<" and record_value < float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            if numeric_filter[idx] == "<=" and record_value <= float(numeric_filter[idx + 1]):
+                all_comparisons.append(True)
+                continue
+            all_comparisons.append(False)
```

Comment:
nit: could be simplified

Reply:
e.g. mapping the operation strings to the corresponding operations. like `>=` to `__ge__()`

Reply:
alternatively using the operator module https://docs.python.org/3/library/operator.html might be an option